tools/llm_bench/llm_bench_utils/ov_utils.py

```diff
@@ -296,15 +297,82 @@ def create_image_gen_model(model_path, device, **kwargs):
+        text_encoder = openvino_genai.CLIPTextModel(model_path / "text_encoder", device.upper(), **ov_config)
+    else:
+        raise RuntimeError(f'==Failure ==: model by path:{model_path} has unsupported text encoder type {text_encoder_type}')
+
+    return text_encoder
+
+
+def get_clip_text_encoder_with_projection(model_index_data, model_path, text_encoder_path, device, ov_config):
+    import openvino_genai
+    text_encoder_type = model_index_data.get(text_encoder_path, [])
+    if ("CLIPTextModelWithProjection" in text_encoder_type):
+        text_encoder = openvino_genai.CLIPTextModelWithProjection(model_path / text_encoder_path, device.upper(), **ov_config)
+    else:
+        raise RuntimeError(f'==Failure ==: model by path:{model_path} has unsupported {text_encoder_path} type {text_encoder_type}')
+
+    return text_encoder
+
+
+def get_unet_model(model_index_data, model_path, device, ov_config):
+    import openvino_genai
+    unet_type = model_index_data.get("unet", [])
+    if ("UNet2DConditionModel" in unet_type):
+        unet = openvino_genai.UNet2DConditionModel(model_path / "unet", device.upper(), **ov_config)
+    else:
+        raise RuntimeError(f'==Failure ==: model by path:{model_path} has unsupported UNet type {unet_type}')
+
+    return unet
+
+
 def create_genai_image_gen_model(model_path, device, ov_config, **kwargs):
     import openvino_genai
 
     adapter_config = get_lora_config(kwargs.get("lora", None), kwargs.get("lora_alphas", []))
     if adapter_config:
         ov_config['adapters'] = adapter_config
 
+    data = {}
+    with open(str(model_path / "model_index.json"), 'r') as f:
+        data = json.load(f)
+
+    model_class_name = data.get("_class_name", "")
+
     start = time.perf_counter()
-    t2i_pipe = openvino_genai.Text2ImagePipeline(model_path, device.upper(), **ov_config)
+
+    vae_type = data.get("vae", [])
+    if ("AutoencoderKL" in vae_type):
+        vae = openvino_genai.AutoencoderKL(model_path / "vae_decoder", device.upper(), **ov_config)
+    else:
+        raise RuntimeError(f'==Failure ==: model by path:{model_path} has unsupported vae decoder type {vae_type}')
+
+    scheduler_type = data.get("scheduler", ["", ""])[1]
+    if (scheduler_type in ["LCMScheduler", "DDIMScheduler", "LMSDiscreteScheduler", "EulerDiscreteScheduler", "FlowMatchEulerDiscreteScheduler"]):
+        scheduler = openvino_genai.Scheduler.from_config(model_path / "scheduler/scheduler_config.json", openvino_genai.Scheduler.Type.AUTO)
+    else:
+        scheduler = openvino_genai.Scheduler.from_config(model_path / "scheduler/scheduler_config.json", openvino_genai.Scheduler.Type.DDIM)
+
+    if model_class_name == "StableDiffusionPipeline":
+        text_encoder = get_clip_text_encoder(data, model_path, device, ov_config)
+        unet = get_unet_model(data, model_path, device, ov_config)
+        t2i_pipe = openvino_genai.Text2ImagePipeline.stable_diffusion(scheduler, text_encoder, unet, vae)
```

Comment:
it's not required to create pipeline from building blocks to change scheduler.
You can create pipeline and then call `pipe.set_scheduler()`

Reply:
creation of the pipeline fails with:
```
t2i_pipe = openvino_genai.Text2ImagePipeline(model_path, device.upper(), **ov_config)
RuntimeError: Exception from ./openvino.genai/src/cpp/src/image_generation/schedulers/types.cpp:59:
Unsupported value for 'scheduler' PNDMScheduler
```

Reply:
can we activate this path only in case if unsupported scheduler required and use default constructor otherwise?
I think it will be better for reduce maintenance efforts if additional pipelines will be added in future

Reply:
thank you, fixed